server/src/main/java/org/elasticsearch/common/network/InetAddresses.java

```diff
@@ -60,6 +64,9 @@ public static boolean isInetAddress(String ipString) {
                     return null;
                 }
             }
+            if (percentIndex != -1) {
```

Comment:
should we make sure that we have something after the `%` in the string and maybe return `null` if `percentIndex == ipString.length() - 1` because that wouldn't be a valid address would it?

Unless you disagree with filtering this case out, could you also add a test for it when adding this check?